app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -42,6 +42,7 @@ def save_inventory_multi(association, hashes, deletes, find_key, child_keys = []
       deletes.reload
     end
     deletes = deletes.to_a
+    deletes_index = deletes.index_by { |x| x }
```

Comment:
It's up to you.

To me, a Set is for when you effectively only have keys but don't really have values.  I find when we try to use a Hash like `{value => value}` or `{value => nil}` only for efficient lookup purposes, that's the perfect time to use a Set.

That being said, I see what you mean about the deletes and the semantics are weird, so might be better to just stick with a Hash in this case.